src/item/Item.php

```diff
@@ -455,6 +457,20 @@ public function getVanillaName() : string{
 		return $this->name;
 	}
 
+	public function getEnchantmentFlag() : int{
+		return $this->enchantmentFlag;
+	}
```

Comment:
I'm not a fan of the name of this function. It's also undocumented.

The purpose of this is to select compatible enchantments to offer, right?

I wonder if adding tags similar to blocks would be a better way to deal with this. The current system doesn't really offer any way to add custom enchantment selection criteria for custom items.

Reply:
I don't know how else to name this function. I'll document it. 
What do you mean by "tags similar to blocks"? You could add an EnchantmentItemFlagRegistry instead of declaring flags in Enchantment, this would allow plugins to edit which items can have certain enchantments applied to them, but that would be a BC break. I don't see any way currently to somehow change the system for determining which items are available for enchantment

Reply:
A tagging system would be more flexible than flags. As for the existing flags properties in Enchantment, I'm not particularly concerned about them - we can just fill those with 0, and have the flags tracked in EnchantmentTableOptionRegistry. This would enable plugins to not only customize which enchantments are generally available, but also which enchantments can apply to which types of item.

Reply:
Although, we could add array constants in EnchantmentTableOptionRegistry that group some tags, for example: 
`public const ARMOR_TAGS = [Tags::HELMET, Tags::CHESTPLATE, Tags::LEGGINGS, Tags::BOOTS];`

And then use some `array_merge` to combine all tags into one array for registration. If that's acceptable, I can do this

Reply:
> > I'm not particularly concerned about them - we can just fill those with 0, and have the tags tracked in EnchantmentTableOptionRegistry
> 
> Yes, but then definitely have to add AnvilEnchantmentRegistry in the future. But I realize that's not a problem.
> 
> Tags may be less convenient than flags from the point of view that when registering the available item tags for enchantment, you will have to specify the tag for each item individually, without grouping. Some enchantments may accept all breakable items, for example, and then you would have to specify tags for all breakable items. And they can also be duplicated. I think flags are better than tags.

This is only a problem if the tags don't allow grouping other tags together.

Reply:
I can't think of a normal grouping system on tags. The problem is that it will not be possible to define composite flags ALL and BREAKABLE in the same way as with flags, without crutches and without a large enumeration of flags